src/main.cc

```diff
@@ -1536,6 +1536,9 @@ SquidMain(int argc, char **argv)
 
 #endif
 
+    Mem::Init();
+    AnyP::UriScheme::Init(); // needs to be before args parsing
```

Comment:
A problem with fixing the underlying problem might be backportability.
This is a live bug on 6.X, a bigger change would make it more difficult to land the fix there.
How about landing a minimal change here, and queueing better fixes for a followup PR?

Reply:
> A problem with fixing the underlying problem might be backportability. 

FWIW, I do not anticipate a proper fix having significant v6 backporting problems.

> This is a live bug on 6.X, a bigger change would make it more difficult to land the fix there.

The "time to land" is not the only factor to consider, and the absolute "time to land" difference is not that great to be an important factor in this decision AFAICT. I do not recommend committing a minimal workaround (that may stop functioning after other seemingly unrelated changes) to v6 just because doing the right thing would take a few extra work hours.

> How about landing a minimal change here, and queueing better fixes for a followup PR?

I do not recommend that strategy in this case (as detailed above), but I will not stand in your way, especially if you commit to follow up with a proper fix before opening other new PRs (to avoid increasing backlog).

However, please note that this PR, in its current shape, is not a minimal change: Removing Mem::Init() call changes is required to make it a minimal change (with the corresponding PR description updates). I would also recommend appending "(workaround)" or a similar suffix to the PR title.